### PR TITLE
Revert "Make torch_geometric models compatible with export (#123403)"

### PR DESCRIPTION
**Pull request by chunyuan-w**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #128377

This reverts commit d78991a7381adb3df5e9b63c365db4506643edce.

This PR reverts https://github.com/pytorch/pytorch/pull/123403 to fix the performance regression as discussed in https://github.com/pytorch/pytorch/issues/127513#issuecomment-2158835653.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]

File: benchmarks/dynamo/torchbench.py [hunk 1]
```diff
@@ -26,20 +26,6 @@
...
-    def state_dict_hook(module, destination, prefix, local_metadata):
-        for name, param in module.named_parameters():
-            if isinstance(destination[name], torch.Tensor) and not isinstance(
-                destination[name], torch.nn.Parameter
-            ):
-                destination[name] = torch.nn.Parameter(destination[name])
-
-    model._register_state_dict_hook(state_dict_hook)
-
-
```
Comment (angelayi) [inline benchmarks/dynamo/torchbench.py:42]: I think we can still keep this change?

**Review by angelayi — APPROVED** [review]

File: benchmarks/dynamo/torchbench.py (hunk 1 above)
Comment (chunyuan-w) [reply benchmarks/dynamo/torchbench.py]: I've added that back and only changed the export related part

**Comment by desertfire** [comment]
Is the idea to cherry-pick this to the 2.4 branch?

**Review by desertfire — APPROVED** [review]

**Comment by chunyuan-w** [comment]
> Is the idea to cherry-pick this to the 2.4 branch?

Yes, I plan to submit a cherry-pick PR to 2.4 when the current PR lands to main.

**Pull request closed without merging**